plugins/inputs/x509_cert/java_key_store.go

```diff
@@ -0,0 +1,90 @@
+package x509_cert
+
+import (
+	"crypto/x509"
+	"fmt"
+	"os"
+
+	"github.com/pavlo-v-chernykh/keystore-go/v4"
+	"software.sslmate.com/src/go-pkcs12"
+)
+
+func (c *X509Cert) processPKCS12(certPath string) ([]*x509.Certificate, error) {
+	data, err := os.ReadFile(certPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read PKCS#12 file: %w", err)
+	}
```

Comment:
You need to cleanup the path here as mentioned in my earlier comment
```suggestion
func normalizePath(path string) string {
	normalized := filepath.ToSlash(path)

	// Removing leading slash in Windows path containing a drive-letter
	// like "file:///C:/Windows/..."
	normalized = strings.TrimreDriveLetter.ReplaceAllString(normalized, "$1")

	return filepath.FromSlash(normalized)
}

func (c *X509Cert) processPKCS12(path string) ([]*x509.Certificate, error) {
	data, err := os.ReadFile(normalizePath(path))
	if err != nil {
		return nil, fmt.Errorf("failed to read PKCS#12 file: %w", err)
	}
```

The same for the JKS implementation.

Reply:
@srebhan the `strings.TrimreDriveLetter.ReplaceAllString` won't exist. Is there a typo here?

```go
❯ go test -v ./plugins/inputs/x509_cert/

# github.com/influxdata/telegraf/plugins/inputs/x509_cert [github.com/influxdata/telegraf/plugins/inputs/x509_cert.test]
plugins/inputs/x509_cert/java_key_store.go:19:23: undefined: strings.TrimreDriveLetter
FAIL	github.com/influxdata/telegraf/plugins/inputs/x509_cert [build failed]
FAIL
```

Reply:
```go
func normalizePath(path string) string {
	normalized := filepath.ToSlash(path)

	// Removing leading slash in Windows path containing a drive-letter
	// like "file:///C:/Windows/..."
	normalized = reDriveLetter.ReplaceAllString(normalized, "$1")

	return filepath.FromSlash(normalized)
}
```

I think you want something like the above.

Pushed, please review.

Reply:
Sorry `strings.TrimreDriveLetter.ReplaceAllString` it was a typing-something else-typing error. :-)